pkg/http/http.go

```diff
@@ -65,19 +71,34 @@ func (h *HTTPGetter) GetWithChecksumAndContext(ctx context.Context, fileURL stri
 		resp.Body.Close()
 		return nil, fmt.Errorf("failed to download %s: %v ", fileURL, fmt.Errorf("status : %v", resp.StatusCode))
 	}
-	return newReadCloserWithChecksum(resp.Body), nil
+	return newReadCloserWithChecksum(resp.Body, checksumDigest)
 }
 
-func newReadCloserWithChecksum(body io.ReadCloser) *readCloserWithChecksum {
-	sha := sha256.New()
-	teeReader := io.TeeReader(body, sha)
-	return &readCloserWithChecksum{body: body, teeReader: teeReader, sha: sha}
+func newReadCloserWithChecksum(body io.ReadCloser, checksumDigest api.ChecksumFormat) (readCloser *readCloserWithChecksum, err error) {
+	var checksumHash hash.Hash
+
+	switch checksumDigest {
+	case api.Sha256:
+		checksumHash = sha256.New()
+	case api.Sha512:
+		checksumHash = sha512.New()
+	default:
+		return nil, fmt.Errorf("unknown checksum digest function %v", checksumDigest)
+	}
+
```

Comment:
Instead of implementing this as a switch, can't you simply pass `sha256.New` or `sha512.New` as arguments to this func and call the passed func here?

Reply:
I've directly created the checksum hasher object in the api.ArtifactDetails. In this way, it can be used directly inside this function. This also allows us to easily use any hasher.